common/types/address.go

```diff
@@ -1,84 +1,120 @@
 package types
 
 import (
-	"encoding/hex"
+	"errors"
 	"fmt"
 	"math/big"
 
+	"github.com/cosmos/btcutil/bech32"
 	"github.com/spacemeshos/go-scale"
 
-	"github.com/spacemeshos/go-spacemesh/common/util"
-	"github.com/spacemeshos/go-spacemesh/hash"
 	"github.com/spacemeshos/go-spacemesh/log"
 )
 
 const (
 	// AddressLength is the expected length of the address.
-	AddressLength = 20
+	AddressLength = 24
+	// AddressReservedSpace define how much bytes from top is reserved in address for future.
+	AddressReservedSpace = 4
 )
 
+var (
+	// ErrWrongAddressLength is returned when the length of the address is not correct.
+	ErrWrongAddressLength = errors.New("wrong address length")
+	// ErrUnsupportedNetwork is returned when a network is not supported.
+	ErrUnsupportedNetwork = errors.New("unsupported network")
+	// ErrDecodeBech32 is returned when an error occurs during decoding bech32.
+	ErrDecodeBech32 = errors.New("error decode to bech32")
+)
+
+// Config is the configuration of the address package.
+type Config struct {
+	NetworkHRP string `mapstructure:"network-hrp"`
+}
+
+var conf = &Config{
+	NetworkHRP: "sm",
+}
+
+// DefaultAddressConfig returns the default configuration of the address package.
+func DefaultAddressConfig() *Config {
+	return conf
+}
+
+// DefaultTestAddressConfig returns the default test configuration of the address package.
+func DefaultTestAddressConfig() *Config {
+	conf = &Config{
+		NetworkHRP: "stest",
+	}
+	return conf
+}
+
 // Address represents the address of a spacemesh account with AddressLength length.
-type Address [AddressLength]byte
+type Address [AddressLength]byte // contains slice 8 bytes unsigned integers
 
-// BytesToAddress returns Address with value b.
-// If b is larger than len(h), b will be cropped from the left.
-func BytesToAddress(b []byte) Address {
-	var a Address
-	a.SetBytes(b)
-	return a
+// FromString returns a new Address from a given string like `sm1abc...`.
+func FromString(src string) (Address, error) {
```

Comment:
it needs to be more specific now, since the module was removed. maybe StringToAddress if there is already BytesToAddress